pkg/server/handlers.go

```diff
@@ -11,10 +13,51 @@ import (
 	"github.com/grafana/iot-sitewise-datasource/pkg/models"
 )
 
+func getErrorResponse(err error, message string) *backend.QueryDataResponse {
+	resp := backend.Responses{
+		"A": backend.DataResponse{
+			Error: errors.New(message + ": " + err.Error()),
+		},
+	}
+	return &backend.QueryDataResponse{Responses: resp}
+}
+
 func processQueries(ctx context.Context, req *backend.QueryDataRequest, handler QueryHandlerFunc) *backend.QueryDataResponse {
 	res := backend.Responses{}
-	for _, v := range req.Queries {
-		res[v.RefID] = handler(ctx, req, v)
+
+	var datasourceJsonData struct {
+		DefaultRegion string `json:"defaultRegion"`
+	}
+
+	defaultRegion := ""
+	if req.PluginContext.DataSourceInstanceSettings != nil &&
+		req.PluginContext.DataSourceInstanceSettings.JSONData != nil {
+
+		if err := json.Unmarshal(req.PluginContext.DataSourceInstanceSettings.JSONData, &datasourceJsonData); err != nil {
+			return getErrorResponse(err, "failed to unmarshal datasource JSON data")
+		}
+		if datasourceJsonData.DefaultRegion != "" {
```

Comment:
Double checking: is this one of the datasources where default region can be "default"? I know it can in cloudwatch